### PR TITLE
V2: Make hpopt config compatible with training config

### PR DESCRIPTION
**Pull request by hwpang**
## Description
https://github.com/chemprop/chemprop/issues/869 points out that the config file generated by hpopt is not compatible with training config. This PR makes it compatible such that a user can use the config file generated by hpopt to train a model. This PR is based on https://github.com/chemprop/chemprop/pull/873 and should be merged after.

## Example / Current workflow
n/a

## Bugfix / Desired workflow
1. Moved the save_config function from train.py to common.py
2. Updated the args using the best params from hpopt
3. Parsed the updated args using TrainSubcommand
4. Saved config file using TrainSubcommand

## Questions
n/a

## Relevant issues
https://github.com/chemprop/chemprop/issues/869
https://github.com/chemprop/chemprop/issues/852

## Checklist
- [x] linted with flake8?
- [x] (if appropriate) unit tests added?

### CODE REVIEW

File: chemprop/cli/hpopt.py [hunk 1]
```diff
@@ -243,6 +244,9 @@ def update_args_with_config(args: Namespace, config: dict) -> Namespace:
             case "init_lr_ratio":
                 setattr(args, "init_lr", value * args.max_lr)
 
+            case "batch_size":
+                setattr(args, "batch_size", int(value))
+
```
Comment (shihchengli) [inline chemprop/cli/hpopt.py:249]:
```suggestion
```

File: chemprop/cli/common.py [hunk 2]
```diff
@@ -84,10 +83,7 @@ def add_common_args(parser: ArgumentParser) -> ArgumentParser:
         "--add-h", action="store_true", help="Whether hydrogens should be added to the mol graph."
     )
     featurization_args.add_argument(
-        "--features-generators",
-        nargs="+",
-        action=LookupAction(MoleculeFeaturizerRegistry),
-        help="Method(s) of generating additional features.",
```
[the remark this answers is not in the document]
Comment (KnathanM) [reply chemprop/cli/common.py:90]: My 876 PR won't be affected by this, but I found a better solution to this problem and opened PR #900.

[the remark this answers is not in the document]
Comment (shihchengli) [reply chemprop/cli/common.py:90]: Per @KnathanM's comment in #900, the `LookupAction` still does some work to populate the choices in `MoleculeFeaturizerRegistry`. If you remove the `LookupAction` here, the program will not complain until running the `main()` in train.py. I think we probably need to add merge this PR after #900.

[the remark this answers is not in the document]
Comment (hwpang) [reply chemprop/cli/common.py:90]: @shihchengli @KnathanM Thanks! I have approved and merged #900. I will merge in the new changes from main.

File: tests/cli/test_cli_regression_mol.py [hunk 3]
```diff
@@ -404,7 +419,7 @@ def test_hyperopt_quick(monkeypatch, data_path, tmp_path):
         "-i",
         input_path,
         "--epochs",
-        "1",
+        "6",
```
Comment (shihchengli) [inline tests/cli/test_cli_regression_mol.py:422]: Can you also change the number of epochs from 1 to 6 in the Optuna test and add the "--search-parameter-keywords", "all", in the optuna test to make them compatible?

**Review by shihchengli — APPROVED** [review]

**Pull request merged**